py/picca/fitter2/chi2.py

```diff
@@ -195,13 +197,34 @@ def fastMC(self):
             d.ico = d.ico/s
             d.cho = cholesky(d.co)
 
+        self.fiducial_values = self.best_fit.values.copy()
+        if len(self.fidfast_mc) != 0:
+            for p in self.fidfast_mc:
+                self.fiducial_values[p] = self.fidfast_mc[p]
+                for d in self.data:
+                    if p in d.par_names:
+                        d.pars_init[p] = self.fidfast_mc[p]
+                        d.par_fixed['fix_'+p] = self.fixfast_mc['fix_'+p]
+
+        self.fiducial_values['SB'] = False
+        for d in self.data:
+            d.fiducial_model = self.fiducial_values['bao_amp']*d.xi_model(self.k, self.pk_lin-self.pksb_lin, self.fiducial_values)
+
+            self.fiducial_values['SB'] = True
+            snl = self.fiducial_values['sigmaNL_per']
+            self.fiducial_values['sigmaNL_per'] = 0
+            d.fiducial_model += d.xi_model(self.k, self.pksb_lin, self.fiducial_values)
+            self.fiducial_values['SB'] = False
+            self.fiducial_values['sigmaNL_per'] = snl
+        del self.fiducial_values['SB']
+
```

Comment:
@mblomqvist, why do you do this? Is it in case a parameter value has changed?

Reply:
That's right